gplearn/functions.py

```diff
@@ -14,6 +14,7 @@
 
 __all__ = ['make_function']
 
+_EPS = np.finfo(np.float64).eps # Used by protected functions
```

Comment:
This would mean that results may differ from machine to machine I guess. Don't really like this!

Reply:
```suggestion
import warnings

_EPS = 1e-6 # or a potentially smaller value
if (np.finfo(np.float32).eps > _EPS):
    message = "This machine's floating point precision is larger than the default "
              "value of epsilon used by gplearn's protected functions."
    warnings.warn(message, UserWarning)
```

This might be too simple of a solution. Notice we check machine epsilon for `np.float32` so that this could be used on 32 bit systems as well. I would also need to remove the explicit conversion to `np.float64` in the definitions of the protected functions.